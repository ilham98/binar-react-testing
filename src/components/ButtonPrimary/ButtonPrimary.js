const ButtonPrimary = (props) => {
  return (
    <button disabled={props.disabled} className="primary">
      Primary Button
    </button>
  );
};

export default ButtonPrimary;
