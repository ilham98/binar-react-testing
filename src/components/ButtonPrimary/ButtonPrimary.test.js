import { render, screen } from "@testing-library/react";
import ButtonPrimary from "./ButtonPrimary";
import renderer from "react-test-renderer";

it("Button rendered correctly", () => {
  render(<ButtonPrimary />);
  const primaryButton = screen.getByRole("button");
  expect(primaryButton).toBeInTheDocument();
});

it("Button has primary class", () => {
  render(<ButtonPrimary />);
  const primaryButton = screen.getByRole("button");
  expect(primaryButton).toHaveClass("primary");
});

it("Snapshot check", () => {
  const tree = renderer.create(<ButtonPrimary />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Disabled primary button", () => {
  render(<ButtonPrimary disabled={true} />);
  const primaryButton = screen.getByRole("button");
  expect(primaryButton).toHaveClass("primary");
  expect(primaryButton).toBeDisabled();
});
