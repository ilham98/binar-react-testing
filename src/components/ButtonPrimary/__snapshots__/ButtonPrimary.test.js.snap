// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot check 1`] = `
<button
  className="primary"
>
  Primary Button
</button>
`;
